### PR TITLE
add phi2 model conversion

### PR DESCRIPTION
**Pull request by guotuofeng**
## Describe your changes
add phi2 model example to convert the microsoft/phi2 model. 

## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: examples/phi2/phi2_optimize.json [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                "dataloader_func": "create_dataloader",
+                "batch_size": 2,
+                "enable_profiling": false
+            }
+        }
+    },
+    "pass_flows": [
+        [
+            "convert"
+        ]
```
Comment (devang-ml) [inline examples/phi2/phi2_optimize.json:89]: Is there a reason to not add perf_tuning in the flow?

File: examples/phi2/phi2_optimize.json [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    },
+    "passes": {
+        "convert_optimum": {
+            "type": "OptimumConversion",
+            "config": {
+                "target_opset": 14,
+                "extra_args": {
+                    "legacy": false,
+                    "no_post_process": false,
+                    "trust_remote_code": true
```
Comment (jambayk) [inline examples/phi2/phi2_optimize.json:64]: "trust_remote_code" should not be here. It's already inherited from the model from_pretrained_args. https://github.com/microsoft/Olive/blob/a013c6057d7b1245e6e048eb8d227be3e2962a84/olive/passes/onnx/optimum_conversion.py#L82

Even if you provided it, it's overwritten by line 82.

File: examples/phi2/phi2_optimize.json (hunk 1 above)
Comment (guotuofeng) [reply examples/phi2/phi2_optimize.json:89]: you are right, the perf_tuning does works when I do benchmark. will add it.
On the other hand, I'd like to add the transformer optimization when ORT support phi2.

File: examples/phi2/phi2_optimize.json (hunk 2 above)
Comment (guotuofeng) [reply examples/phi2/phi2_optimize.json:64]: it seems the extra_args will take precedence over the pretrained args. I add it here just for explicit purpose. the optimum conversion doesn't used from the time being. I will remove it for simplicity.

File: examples/phi2/phi2_optimize.json [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                                }
+                            }
+                        }
+                    }
+                }
+            ]
+        }
+    },
+    "passes": {
+        "convert_optimum": {
```
Comment (trajepl) [inline examples/phi2/phi2_optimize.json:57]: Both of optimum and onnx conversion work here? Should we remove this? Seems you did not use that in pass flows.

Comment (trajepl) [reply examples/phi2/phi2_optimize.json:57]: If you want to keep it, should we set the external saving config as following?
"save_as_external_data": true,
"all_tensors_to_one_file": true

Comment (guotuofeng) [reply examples/phi2/phi2_optimize.json:57]: the two configs are removed already for optimum conversion in https://github.com/microsoft/Olive/pull/851

**Review by jambayk — APPROVED** [review]

File: examples/phi2/readme.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+## Prerequisites
+* einops
+
+## Usage
+```bash
+python -m olive.workflows.run --config phi2_optimize.json
+```
+
+## Limitations
+From the time being, the official [phi2](https://huggingface.co/microsoft/phi-2) model could not be exported to ONNX by using the official model code. Therefore, we need patch the forward method to do preprocessing and postprocessing for the past_key_values arguments. When the official model could be exported to ONNX, we will remove this patch.
```
Comment (jambayk) [inline examples/phi2/readme.md:10]: This is not needed anymore?

Comment (guotuofeng) [reply examples/phi2/readme.md:10]: good catch

**Review by jambayk — APPROVED** [review]

**Pull request merged**